packages/transport-webrtc/src/webrtc/rtc-data-channel.ts

```diff
@@ -0,0 +1,142 @@
+import { fromString as uint8ArrayFromString } from 'uint8arrays/from-string'
+import type node from 'node-datachannel'
+
+export class DataChannel extends EventTarget implements RTCDataChannel {
+  binaryType: BinaryType
+
+  readonly maxPacketLifeTime: number | null
+  readonly maxRetransmits: number | null
+  readonly negotiated: boolean
+  readonly ordered: boolean
+
+  onbufferedamountlow: ((this: RTCDataChannel, ev: Event) => any) | null
+  onclose: ((this: RTCDataChannel, ev: Event) => any) | null
+  onclosing: ((this: RTCDataChannel, ev: Event) => any) | null
+  onerror: ((this: RTCDataChannel, ev: Event) => any) | null
+  onmessage: ((this: RTCDataChannel, ev: MessageEvent) => any) | null
+  onopen: ((this: RTCDataChannel, ev: Event) => any) | null
+
+  #dataChannel: node.DataChannel
+  #bufferedAmountLowThreshold: number
+  #readyState: RTCDataChannelState
+
+  constructor (dataChannel: node.DataChannel) {
+    super()
+
+    this.#dataChannel = dataChannel
+    this.#readyState = 'connecting'
+
+    // TODO: what is the default?
+    this.#bufferedAmountLowThreshold = 0
+
+    this.binaryType = 'arraybuffer'
+
+    this.#dataChannel.onOpen(() => {
+      this.#readyState = 'open'
+      this.dispatchEvent(new Event('open'))
+    })
+    this.#dataChannel.onClosed(() => {
+      this.#readyState = 'closed'
+      this.dispatchEvent(new Event('close'))
+    })
+    this.#dataChannel.onError((msg) => {
+      this.#readyState = 'closed'
+      this.dispatchEvent(new RTCErrorEvent('error', {
+        error: new RTCError({
+          errorDetail: 'data-channel-failure'
+        }, msg)
+      }))
+    })
+    this.#dataChannel.onBufferedAmountLow(() => {
+      this.dispatchEvent(new Event('bufferedamountlow'))
+    })
+    this.#dataChannel.onMessage((data: string | Uint8Array) => {
+      if (typeof data === 'string') {
+        data = uint8ArrayFromString(data)
+      }
+
+      this.dispatchEvent(new MessageEvent('message', { data }))
+    })
+
+    // forward events to properties
+    this.addEventListener('message', event => {
+      this.onmessage?.(event as MessageEvent<ArrayBuffer>)
+    })
+    this.addEventListener('bufferedamountlow', event => {
+      this.onbufferedamountlow?.(event)
+    })
+    this.addEventListener('error', event => {
+      this.onerror?.(event)
+    })
+    this.addEventListener('close', event => {
+      this.onclose?.(event)
+    })
+    this.addEventListener('closing', event => {
+      this.onclosing?.(event)
+    })
+    this.addEventListener('open', event => {
+      this.onopen?.(event)
+    })
+
+    this.onbufferedamountlow = null
+    this.onclose = null
+    this.onclosing = null
+    this.onerror = null
+    this.onmessage = null
+    this.onopen = null
+
+    // TODO: update these
```

Comment:
Missing the setters for these.

Reply:
These are readonly properties according to the spec.